core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -96,34 +97,43 @@ trait MethodConfigurationComponent {
     /*
       the core methods
      */
-    def save(workspaceContext: SlickWorkspaceContext, methodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = {
 
-      def saveMaps(configId: Long) = {
-        val prerequisites = methodConfig.prerequisites.map { case (key, value) => marshalConfigPrereq(configId, key, value) }
-        val inputs = methodConfig.inputs.map { case (key, value) => marshalConfigInput(configId, key, value) }
-        val outputs = methodConfig.outputs.map{ case (key, value) => marshalConfigOutput(configId, key, value) }
-
-        (methodConfigurationPrereqQuery ++= prerequisites) andThen
-          (methodConfigurationInputQuery ++= inputs) andThen
-            (methodConfigurationOutputQuery ++= outputs)
-      }
+    // the save method is used for both creating a new method config and updating an edited method config
+    // the save method is not used for renaming a method config, see the rename method for that
+    def save(workspaceContext: SlickWorkspaceContext, newMethodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = {
 
-      uniqueResult[MethodConfigurationRecord](findByName(workspaceContext.workspaceId, methodConfig.namespace, methodConfig.name)) flatMap {
+      uniqueResult[MethodConfigurationRecord](findByName(workspaceContext.workspaceId, newMethodConfig.namespace, newMethodConfig.name)) flatMap {
         case None =>
-          val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) +=  marshalMethodConfig(workspaceContext.workspaceId, methodConfig))
+          val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) +=  marshalMethodConfig(workspaceContext.workspaceId, newMethodConfig))
           configInsert flatMap { configId =>
-            saveMaps(configId)
+            saveMaps(newMethodConfig, configId)
           }
         case Some(methodConfigRec) =>
```

Comment:
Nitpick: can you rename this to "current" or "existing" methodConfigRec? To contrast with "new"?